sdk/go/pulumi-language-go/main.go

```diff
@@ -190,15 +189,23 @@ func (host *goLanguageHost) GetRequiredPlugins(ctx context.Context,
 		return nil, errors.Wrap(err, "couldn't find go binary")
 	}
 
+	cmd := exec.Command(gobin, "version")
+	stdout, err := cmd.Output()
+	if err != nil {
+		return nil, fmt.Errorf("failed to determine go version: %w", err)
+	}
+	rawVersion := string(stdout)
+	if err = checkMinimumGoVersion(rawVersion); err != nil {
+		return nil, err
+	}
+
 	// don't wire up stderr so non-module users don't see error output from list
-	cmd := exec.Command(gobin, "list", "-m", "-json", "-mod=mod", "all")
+	cmd = exec.Command(gobin, "list", "-m", "-json", "-mod=mod", "all")
 	cmd.Env = os.Environ()
-
-	stdout, err := cmd.Output()
+	stdout, err = cmd.Output()
 	if err != nil {
-		// will err if the project isn't using modules
-		logging.V(5).Infof("GetRequiredPlugins: Error discovering plugin requirements: %s", err.Error())
-		return &pulumirpc.GetRequiredPluginsResponse{}, nil
+		logging.V(5).Infof("GetRequiredPlugins: Error discovering plugin requirements using go modules: %s", err.Error())
```

Comment:
Previously returning an empty response essentially silently disabled plugin acquisition for users with glide or dep projects. We can reinstate the behavior if we feel we aren't ready for this breaking change.